### PR TITLE
cluster: use systemd cgroup driver for cgroupv2

### PR DESCRIPTION
**Pull request by bobbypage**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

* Detect if image is using cgroupv2
* Configure both kubelet and containerd to use systemd cgroup driver
  when running under cgroupv2

Systemd cgroup driver is recommended to be used when running on
cgroupv2. It is also the default in moby
https://github.com/moby/moby/pull/40846

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -3060,6 +3071,13 @@ EOF
       cni_template_path=""
     fi
   fi
+
+   # Use systemd cgroup driver when running on cgroupv2
+  local systemdCgroup="false"
+  if [[ "${CGROUP_CONFIG-}" == "cgroup2fs" ]]; then
```
Comment (cheftako) [inline cluster/gce/gci/configure-helper.sh:3077]: Might be good to add a default value in config-default.sh

Comment (bobbypage) [reply cluster/gce/gci/configure-helper.sh:3077]: I thought about it, but I don't think it makes sense in this context.

The issue is that we want to default to the systemd cgroup driver when running on cgroupv2 OS image. `config-default.sh` sets defaults ahead of time for deployment. At that time, we don't know ahead if the OS image will be cgroupv2 enabled or not.

As a result, this check is done when the VM actually boots which is when we can detect the cgroup configuration. Let me know what you think and if that makes sense.

**Pull request merged**